### PR TITLE
Support resource type's flag

### PR DESCRIPTION
**Pull request by ms-henglu**
And update bicep types to ea703e2aba0d1c024f33124ee2cd34bc0c6084b5

### CODE REVIEW

File: internal/langserver/handlers/tfschema/candidates.go [hunk 1]
```diff
@@ -60,10 +60,24 @@ func valueCandidates(values []string, r lsp.Range, isOrdered bool) []lsp.Complet
...
+}
+
+func typeCandidates(prefix *string, r lsp.Range, allowReadOnly bool) []lsp.CompletionItem {
 	candidates := make([]lsp.CompletionItem, 0)
 	if prefix == nil || !strings.Contains(*prefix, "@") {
-		for resourceType := range azure.GetAzureSchema().Resources {
+		for resourceType, resource := range azure.GetAzureSchema().Resources {
+			if len(resource.Definitions) > 0 {
+				def, err := resource.Definitions[0].GetDefinition()
+				if err == nil && def.IsReadOnly() && !allowReadOnly {
```
Comment (magodo) [inline internal/langserver/handlers/tfschema/candidates.go:77]: What if `err` is not nil here?

File: internal/langserver/handlers/tfschema/candidates.go [hunk 2]
```diff
@@ -60,10 +60,24 @@ func valueCandidates(values []string, r lsp.Range, isOrdered bool) []lsp.Complet
...
+func dataSourceTypeCandidates(prefix *string, r lsp.Range) []lsp.CompletionItem {
+	return typeCandidates(prefix, r, true)
+}
+
+func typeCandidates(prefix *string, r lsp.Range, allowReadOnly bool) []lsp.CompletionItem {
 	candidates := make([]lsp.CompletionItem, 0)
 	if prefix == nil || !strings.Contains(*prefix, "@") {
-		for resourceType := range azure.GetAzureSchema().Resources {
+		for resourceType, resource := range azure.GetAzureSchema().Resources {
+			if len(resource.Definitions) > 0 {
```
Comment (magodo) [inline internal/langserver/handlers/tfschema/candidates.go:75]: Will there be case that the length > 1?

**Review by magodo — COMMENTED** [review]
I've left some minor comments, other than that this LGTM!

File: internal/langserver/handlers/tfschema/candidates.go (hunk 2 above)
Comment (ms-henglu) [reply internal/langserver/handlers/tfschema/candidates.go:75]: Yes, but it's not necessary to check all definitions, it's enough to take the first one.

File: internal/langserver/handlers/tfschema/candidates.go (hunk 1 above)
Comment (ms-henglu) [reply internal/langserver/handlers/tfschema/candidates.go:77]: Normally err is always nil here, but if err is not nil, then we can't know whether this type is readonly or not, so skip checking on it.

**Review by magodo — APPROVED** [review]
LGTM!

**Pull request merged**